### ISSUE TITLE
new figsize is bad for subplots with fontsize 12

### ISSUE THREAD

**Issue by efiring** [matplotlib/matplotlib#6495]
Classic: figsize (8, 6)
v2.x: figsize (6.4, 4.8)
fontsize: 12 pts in both cases

Together with the infamous outward ticks, this change in figsize wreaks havoc on label placement and axes positions.  For a single default subplot, possibly adequate adjustments have been made, but the situation for multiple subplots looks hopeless.  Here is the default 4-panel figure, without even trying to add any labels or titles:
![subplots4](https://cloud.githubusercontent.com/assets/85125/15630789/520aed78-24e5-11e6-928d-c5486e12e086.png)
Possible solutions:
1) Restore the old figure size.
2) Reduce the base font size
In either case, of course, all positioning parameters will need to be reviewed and adjusted.
I omitted one option: I think that increasing the subplotparams wspace and hspace enough to make 4 subplots look reasonable would leave too little room for the plots themselves.

I think the change to figsize is also responsible for many broken gallery examples as well.  

The rationale for reducing the figsize was that (8, 6) is too large for most publication uses, and that with the increase in figure.dpi to 100 to match savefig.dpi, it is also too large for the screen, making it hard to work with multiple figures, and, depending on the user's screen, possibly even triggering resizing, which destroys any attempt to design a layout with fixed dimensions.

Therefore I think that the best option might be the more painful one: drop fontsize from 12 to 10, and adjust everything else accordingly.

**Comment by efiring** [comment]
I'm now convinced that building the parameters around the reduced fontsize is the best way to go, and actually likely to be less painful than the alternatives.
In addition, I think that if we keep the outward ticks, then we should turn off the top and right ticks.  That will reduce the ugliness and loss of useful space, especially when there is more than one subplot.

**Comment by tacaswell** [comment]
We nominally have some data about this from the analytic on the website which sent me down a kinda fun analysis rabbit hole.

caveat: browsers sometimes lie (see http://whatsmy.browsersize.com/ firefox gets my screens correct, chromium is internally doubling to account for the high resolution screen and reports 1/2 the correct screen size) and this also include the small amount of mobile traffic we get.  This is based on data from Mach 2015 till now.

![so](https://cloud.githubusercontent.com/assets/199813/15635296/c12a9db6-25a8-11e6-949c-673b21b50221.png)
![so2](https://cloud.githubusercontent.com/assets/199813/15635304/17cc426e-25a9-11e6-9fea-0a70233dd31f.png)

(click the images to see big versions)

If we assume 1280x768 that as a screen size that catches 94.8% of our user base.  If we say a figure should take up no more that 1/4 of the screen by default, then our target resolution should be 640x384.  Currently our default figure size and dpi ((6.4, 4.8), 100) gives 640x480 which is sort of close (it is the difference between a 4:3 and 5:3 aspect ratio) but is enough that only ~50% of our users can tile 2 figures vertically (and that is not considering the GUI/window manager space).

I think reducing the font size is the right thing to do.  

When thinking about font sizes what really ends up mattering is the ratio between the font height (in pixels) and the figure height (in pixels).  Once the image gets embedded in something else it almost always get scaled uniformly (ex a figure that fills a slide in a presentation) so the font size in the figure is not strictly related to the font size in the rest of the document.  When we reduced the figure size in inches to restore it's size in pixels, we increased the font size relative to the figure size, thus we need to shrink the font a bit to get back to the previous ratio.


Related:

https://www.w3counter.com/globalstats.php?year=2016&month=4 gives screen resolution statistics for the general web user.

----

```python
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from io import StringIO
import re

inp = StringIO('''Screen Resolution,Sessions,% New Sessions,New Users,Bounce Rate,Pages / Session,Avg. Session Duration,Goal Conversion Rate,Goal Completions,Goal Value
1920x1080,"1,958,104",27.54%,"539,352",49.94%,3.01,00:05:10,0.00%,0,$0.00
1366x768,"1,269,788",34.49%,"438,006",51.80%,2.87,00:04:51,0.00%,0,$0.00
1440x900,"834,959",26.92%,"224,773",50.59%,2.87,00:05:04,0.00%,0,$0.00
1280x800,"722,484",27.39%,"197,884",51.19%,2.80,00:05:06,0.00%,0,$0.00
1920x1200,"657,828",21.96%,"144,475",49.50%,2.98,00:05:21,0.00%,0,$0.00
1680x1050,"483,351",25.25%,"122,032",49.74%,2.99,00:05:20,0.00%,0,$0.00
1600x900,"393,355",31.16%,"122,574",50.97%,2.95,00:05:00,0.00%,0,$0.00
1280x1024,"353,511",29.62%,"104,718",50.28%,3.05,00:05:15,0.00%,0,$0.00
2560x1440,"303,764",21.50%,"65,323",49.91%,2.91,00:05:18,0.00%,0,$0.00
1280x720,"93,364",31.96%,"29,837",51.64%,2.86,00:04:56,0.00%,0,$0.00
1536x864,"79,725",32.83%,"26,177",50.14%,2.92,00:05:11,0.00%,0,$0.00
1600x1200,"65,708",23.65%,"15,537",49.32%,3.07,00:05:25,0.00%,0,$0.00
2560x1600,"55,616",21.79%,"12,119",49.70%,2.96,00:05:11,0.00%,0,$0.00
1024x768,"51,606",57.15%,"29,495",62.15%,2.62,00:03:37,0.00%,0,$0.00
360x640,"50,188",60.04%,"30,134",65.56%,2.27,00:02:29,0.00%,0,$0.00
768x1024,"37,930",51.76%,"19,632",59.98%,2.90,00:03:15,0.00%,0,$0.00
1360x768,"33,322",36.22%,"12,070",51.73%,2.97,00:04:55,0.00%,0,$0.00
320x568,"20,236",63.93%,"12,937",72.18%,1.82,00:01:44,0.00%,0,$0.00
375x667,"19,566",62.64%,"12,257",70.84%,1.96,00:01:54,0.00%,0,$0.00
2048x1152,"18,575",23.13%,"4,297",49.98%,2.96,00:05:15,0.00%,0,$0.00
1200x1920,"17,729",19.33%,"3,427",51.01%,2.87,00:05:19,0.00%,0,$0.00
1080x1920,"17,538",22.81%,"4,000",50.30%,3.16,00:05:20,0.00%,0,$0.00
1440x960,"14,178",33.04%,"4,684",52.33%,2.94,00:04:48,0.00%,0,$0.00
2560x1080,"12,294",25.58%,"3,145",50.28%,2.94,00:05:02,0.00%,0,$0.00
1536x960,"10,540",27.93%,"2,944",48.33%,3.20,00:05:40,0.00%,0,$0.00
1024x600,"10,367",44.90%,"4,655",53.70%,2.79,00:04:54,0.00%,0,$0.00
3840x2160,"8,028",26.26%,"2,108",50.29%,2.99,00:05:04,0.00%,0,$0.00
1600x1000,"6,811",15.87%,"1,081",48.50%,3.08,00:05:27,0.00%,0,$0.00
1344x840,"6,800",29.94%,"2,036",47.34%,3.32,00:05:56,0.00%,0,$0.00
1760x990,"6,549",7.63%,500,48.54%,2.82,00:05:46,0.00%,0,$0.00
1280x768,"6,397",38.66%,"2,473",52.49%,3.08,00:04:41,0.00%,0,$0.00
1024x1280,"6,313",19.39%,"1,224",47.27%,3.18,00:06:07,0.00%,0,$0.00
1050x1680,"5,820",19.04%,"1,108",48.44%,3.06,00:05:52,0.00%,0,$0.00
3440x1440,"5,705",25.35%,"1,446",49.78%,3.09,00:05:20,0.00%,0,$0.00
414x736,"5,454",58.86%,"3,210",70.33%,2.07,00:02:08,0.00%,0,$0.00
320x480,"5,290",58.90%,"3,116",69.89%,1.93,00:02:22,0.00%,0,$0.00
1280x960,"5,255",37.68%,"1,980",51.08%,3.14,00:05:07,0.00%,0,$0.00
1440x810,"5,058",15.78%,798,50.43%,2.92,00:05:46,0.00%,0,$0.00
2400x1350,"4,949",19.82%,981,51.69%,2.87,00:05:20,0.00%,0,$0.00
1093x614,"4,603",43.54%,"2,004",50.66%,2.99,00:04:43,0.00%,0,$0.00
2133x1200,"4,491",24.20%,"1,087",51.46%,2.97,00:05:04,0.00%,0,$0.00
(not set),"4,330",96.37%,"4,173",88.27%,1.13,00:00:33,0.00%,0,$0.00
2144x1206,"4,100",7.90%,324,48.80%,2.76,00:05:16,0.00%,0,$0.00
1829x1029,"4,089",30.96%,"1,266",49.55%,3.50,00:04:58,0.00%,0,$0.00
1152x864,"4,060",39.80%,"1,616",55.32%,2.90,00:04:22,0.00%,0,$0.00
1472x828,"4,052",10.69%,433,47.63%,2.98,00:05:46,0.00%,0,$0.00
800x600,"3,852",72.74%,"2,802",64.23%,2.44,00:02:07,0.00%,0,$0.00
1200x1600,"3,835",18.83%,722,49.60%,3.07,00:05:46,0.00%,0,$0.00
1400x1050,"3,768",32.86%,"1,238",47.27%,3.26,00:05:23,0.00%,0,$0.00
1252x704,"3,694",6.96%,257,48.16%,2.69,00:05:31,0.00%,0,$0.00
1760x1100,"3,640",4.15%,151,49.84%,2.69,00:05:30,0.00%,0,$0.00
1024x819,"3,632",30.34%,"1,102",48.79%,2.90,00:04:59,0.00%,0,$0.00
1138x640,"3,552",16.39%,582,48.17%,2.99,00:05:37,0.00%,0,$0.00
1364x768,"3,460",30.23%,"1,046",52.69%,2.73,00:04:39,0.00%,0,$0.00
1745x982,"3,391",31.29%,"1,061",49.45%,3.50,00:05:26,0.00%,0,$0.00
1376x774,"3,316",12.30%,408,48.31%,2.91,00:05:44,0.00%,0,$0.00
1706x960,"3,253",25.58%,832,48.69%,3.02,00:05:15,0.00%,0,$0.00
412x732,"2,919",67.04%,"1,957",66.12%,2.23,00:02:12,0.00%,0,$0.00
1368x768,"2,910",39.04%,"1,136",50.48%,3.06,00:05:01,0.00%,0,$0.00
1368x912,"2,895",36.89%,"1,068",49.60%,3.02,00:04:47,0.00%,0,$0.00
1670x939,"2,859",33.05%,945,49.49%,3.61,00:04:49,0.00%,0,$0.00
1400x875,"2,760",11.52%,318,49.89%,2.91,00:05:46,0.00%,0,$0.00
1120x700,"2,735",17.55%,480,47.97%,3.07,00:05:56,0.00%,0,$0.00
1301x731,"2,654",42.95%,"1,140",50.04%,3.19,00:04:34,0.00%,0,$0.00
720x1280,"2,585",57.21%,"1,479",68.24%,2.32,00:02:41,0.00%,0,$0.00
2021x1137,"2,556",27.86%,712,51.10%,3.12,00:04:53,0.00%,0,$0.00
1707x960,"2,551",23.72%,605,51.98%,2.81,00:05:00,0.00%,0,$0.00
1525x858,"2,511",7.49%,188,50.54%,2.88,00:06:06,0.00%,0,$0.00
320x534,"2,471",59.98%,"1,482",66.37%,2.06,00:02:46,0.00%,0,$0.00
1152x720,"2,426",31.12%,755,50.49%,3.03,00:05:05,0.00%,0,$0.00
1438x808,"2,387",36.87%,880,50.78%,3.08,00:04:25,0.00%,0,$0.00
1440x2560,"2,368",22.59%,535,50.80%,2.77,00:05:00,0.00%,0,$0.00
1120x630,"2,334",8.18%,191,49.10%,2.60,00:05:39,0.00%,0,$0.00
1333x750,"2,322",16.54%,384,51.38%,2.93,00:05:23,0.00%,0,$0.00
800x1280,"2,249",47.22%,"1,062",56.56%,3.61,00:04:02,0.00%,0,$0.00
360x592,"2,177",61.14%,"1,331",65.78%,2.50,00:02:25,0.00%,0,$0.00
640x360,"2,141",60.72%,"1,300",60.44%,2.82,00:02:54,0.00%,0,$0.00
1422x800,"2,135",32.18%,687,50.12%,3.23,00:05:02,0.00%,0,$0.00
1600x2560,"2,045",20.29%,415,52.37%,2.60,00:05:13,0.00%,0,$0.00
1371x771,"2,029",28.93%,587,46.38%,3.40,00:05:37,0.00%,0,$0.00
3200x1800,"2,019",27.54%,556,53.10%,2.74,00:04:41,0.00%,0,$0.00
600x960,"1,956",48.01%,939,59.00%,2.98,00:02:51,0.00%,0,$0.00
1242x698,"1,942",39.80%,773,50.72%,3.17,00:04:41,0.00%,0,$0.00
960x600,"1,933",36.52%,706,54.53%,3.06,00:04:26,0.00%,0,$0.00
384x640,"1,915",54.20%,"1,038",72.06%,2.10,00:02:07,0.00%,0,$0.00
1920x975,"1,896",34.81%,660,52.16%,2.65,00:05:15,0.00%,0,$0.00
3840x1200,"1,875",19.09%,358,47.09%,3.22,00:06:02,0.00%,0,$0.00
1472x920,"1,857",7.75%,144,45.34%,3.08,00:06:04,0.00%,0,$0.00
1745x981,"1,795",24.01%,431,51.14%,2.60,00:04:47,0.00%,0,$0.00
1477x831,"1,781",34.53%,615,49.35%,3.25,00:04:35,0.00%,0,$0.00
2048x1280,"1,719",23.44%,403,48.87%,2.94,00:05:14,0.00%,0,$0.00
2144x1340,"1,706",7.62%,130,47.25%,3.00,00:05:59,0.00%,0,$0.00
480x800,"1,693",61.96%,"1,049",67.87%,1.93,00:02:15,0.00%,0,$0.00
1540x963,"1,683",5.11%,86,49.14%,2.65,00:05:28,0.00%,0,$0.00
1829x1143,"1,672",24.82%,415,47.73%,3.46,00:05:36,0.00%,0,$0.00
1067x853,"1,639",13.79%,226,43.62%,3.37,00:06:10,0.00%,0,$0.00
601x962,"1,603",47.41%,760,57.83%,3.14,00:03:55,0.00%,0,$0.00
1024x640,"1,579",29.01%,458,51.30%,3.46,00:05:04,0.00%,0,$0.00
2880x1620,"1,568",25.64%,402,52.23%,2.94,00:05:18,0.00%,0,$0.00
1500x1000,"1,525",31.80%,485,49.38%,3.05,00:05:39,0.00%,0,$0.00
,"8,041,577",28.93%,"2,326,365",50.92%,2.93,00:05:03,0.00%,0,$0.00''')


# load data
def clean_ga_data(inp):
    '''Clean up csv data from google analytics

    Parameters
    ----------
    str : inp
        The multi-line string of the resolution table from GA.

    Returns
    -------
    DataFrame
        Columns ['Sessions', 'New Users', 'width', 'height'].  Indexed on
        'Screen Resolution', the 'WxH' string.


    '''
    df = pd.read_csv(inp, sep=",")
    df = df[['Screen Resolution', 'Sessions', 'New Users']]

    # drop (not set) and np.nan rows
    df = df.loc[df['Screen Resolution'].apply(
        lambda x: re.match('[0-9]+x[0-9]+', str(x)) is not None)]

    # why do integers have commas in them?
    for col in ['Sessions', 'New Users']:
        df[col] = df[col].str.replace(',', '').astype(int)

    df_shape = df['Screen Resolution'].str.split('x', expand=True).astype(int)
    df_shape.columns = ['width', 'height']
    df = pd.concat([df, df_shape], axis=1)
    df = df.set_index(['Screen Resolution'])

    return df


def atleast_fraction(df, dim):
    sorted_df = df.sort_values(dim)
    for col in ['Sessions', 'New Users']:
        cs = sorted_df[col].cumsum()
        cs /= cs.max()
        sorted_df['{} at least {}'.format(col, dim)] = 1 - cs
    return sorted_df


def plot_atleast_fraction(df, dim, ax):
    atl = atleast_fraction(df, dim)
    out = []
    for col in ['Sessions', 'New Users']:
        ln, = ax.plot(dim, '{} at least {}'.format(col, dim),
                      data=atl, drawstyle='steps-pre')
        out.append(ln)
    ax.legend()
    ax.set_xlabel('screen {} [px]'.format(dim))
    ax.set_ylabel('fraction at least this big')


def compute_size_heatmap(df, col):
    max_size = tuple(df[['height', 'width']].max())
    im = np.zeros(max_size)
    norm = df[col].sum()

    for r in df.iterrows():
        r = r[1]
        w, h = r['width'], r['height']
        im[:(h+1), :(w+1)] += r[col]
    im /= norm
    return im


def show_size_heat_map(df, col, ax):
    im = compute_size_heatmap(df, col)
    ax.set_title(col)
    ax.set_xlabel('screen width [px]')
    ax.set_ylabel('screen height [px]')
    return ax.imshow(im)

df = clean_ga_data(inp)

fig_single, (ax_w, ax_h) = plt.subplots(1, 2, figsize=(16, 8))
plot_atleast_fraction(df, 'height', ax_h)
plot_atleast_fraction(df, 'width', ax_w)

fig, ax = plt.subplots(figsize=(12, 6))
im = show_size_heat_map(df, 'Sessions', ax)
im.set_clim(.8, 1)
fig.colorbar(im, ax=ax, extend='min')
ax.set_ylim(1080, 0)
ax.set_xlim(0, 1920)

```

**Comment by tacaswell** [comment]
@efiring If you are working on this I will work on #6380

**Comment by efiring** [comment]
OK.  Here is the result of 10 pt font with top and RH ticks turned off, for comparison with the original minimal example above.  Big improvement, I think.  Is it OK to proceed with this strategy of leaving out the top and RH ticks?  With outward ticks, I think it is essential.  Users who want more alignment cues will simply have to turn on the grid.
![subplots_10pt](https://cloud.githubusercontent.com/assets/85125/15635580/ca74e4ec-257d-11e6-8a6b-53fc109578ae.png)

**Comment by tacaswell** [comment]
Despite it being a much bigger change, I think that looks better than out on all sides.

**Comment by efiring** [comment]
Good!  I am going out for a couple hours, but I can work on this when I return.

**Comment by afvincent** [comment]
If I may, when using outward ticks, using them only on the right and the bottom spines (as Eric proposes) seems to be rather usual among other plotting softwares (at least in their gallery...). See for example:
[Origin ex 1](http://cloud.originlab.com/www/resources/graph_gallery/images_galleries_new/Panel_Graph_with_Inset_Plots_High_Resolution_Electron_Energyloss_Spectra.gif)
[Origin ex 2](http://cloud.originlab.com/www/resources/graph_gallery/images_galleries_new/Symbol_Plot_Color_Transparency_v5.png)
[Igor ex 1](https://www.wavemetrics.com/products/igorpro/gallery/user_jurkus.htm)
[Igor ex 2](https://www.wavemetrics.com/products/igorpro/gallery/user_nmr.htm)
[SigmaPlot ex (1st example of the showcase)](http://www.sigmaplot.co.uk/products/sigmaplot/graph-showcase.php)

**Comment by afvincent** [comment]
If font height relative to figure height is really what matters, then switching to a 10 pt font for the new default smaller--*in inches*--figure seems to be what the math says: 12 pt * (6.4 in / 8 in) = 9.6 pt

**Comment by efiring** [comment]
closed by #6500
